mindsdb/api/http/initialize.py

```diff
@@ -216,7 +216,17 @@ def root_index(path):
                 'Not found',
                 'The endpoint you are trying to access does not exist on the server.'
             )
-        if static_root.joinpath(path).is_file():
+
+        # Check for directory path traversal.
+        full_path = os.path.normpath(os.path.join(static_root, path))
+        if not full_path.startswith(str(static_root)):
```

Comment:
Better to use pathlib.Path. Something like
```python
if not (static_path / path).resolve().is_relative_to(static_path):
```

Reply:
Hey @StpMax,
From what I have understood, it looks like the `is_relative_to` method is only available in Python 3.10 and above as explained in this [issue](https://github.com/Significant-Gravitas/AutoGPT/issues/2027). Should we still make this change?

Reply:
Hey @StpMax,
Sorry for the delay in getting to this, but can you please check if the change I've made is in line with what you have advised?